### DISCUSSION TITLE
Problem when Compile rheoTool

### DISCUSSION THREAD

**Discussion by wancha0125** [fppimenta/rheoTool#104] [Q&A]
Hi fppimenta ,

May you kindly assist me. I successfully install OpenFOAM9 Eigen and Petsc, but fails to compile rheoTool.
 Here is the error infor. Thanks :).

^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from rheoFoam.C:43:
../shared/singlePhase/CorrectPhi.H:3:1: error: ‘phi’ does not name a type
    3 | phi = mesh.Sf() & Uf();
      | ^~~
../shared/singlePhase/CorrectPhi.H:5:16: error: expected constructor, destructor, or type conversion before ‘(’ token
    5 | correctUphiBCs (U, phi, true);
      |                ^
../shared/singlePhase/CorrectPhi.H:8:1: error: expected constructor, destructor, or type conversion before ‘(’ token
    8 | (
      | ^
In file included from ../shared/singlePhase/CorrectPhi.H:17,
                 from rheoFoam.C:43:
/home/dzh/OpenFOAM/OpenFOAM-9/src/finiteVolume/lnInclude/continuityErrs.H:32:1: error: expected unqualified-id before ‘
{’ token
   32 | {
      | ^
In file included from rheoFoam.C:43:
../shared/singlePhase/CorrectPhi.H:20:18: error: expected constructor, destructor, or type conversion before ‘(’ token
   20 | fvc::makeRelative(phi, U);
      |                  ^
In file included from /home/dzh/OpenFOAM/OpenFOAM-9/src/OpenFOAM/lnInclude/postProcess.H:105,
                 from rheoFoam.C:58:
./createFields.H: In function ‘int main(int, char**)’:
./createFields.H:33:1: error: ‘pressureReference’ was not declared in this scope
   33 | pressureReference pressureReference(p, mesh.solutionDict().subDict("SIMPLE"));
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

**Comment by yamin02** [comment]
Any idea how this is solved ?

**Reply by EdgarOrtegaRoano** [reply]
I believe it's because the linker found a version of CorrectPhi.H which is not the appropriate one, since there are a few files called like that.
Maybe you should modify 
#include CorrectPhi.H   (in rheoFoam.C and similars)
to
#include "/opt/openfoam9/src/finiteVolume/cfdTools/general/CorrectPhi/CorrectPhi.H"
if that is the correct file to point to?
